CONTRIBUTING.md

```diff
@@ -6,6 +6,212 @@ This guide explains how to contribute to cocotb, and documents the processes we
+Finally, they should notify the maintainers that they are able and willing to accept questions.
+
+To be able to add labels and close issues and PRs you will need special permissions.
+Contact a [maintainer](#maintainer) if you are interested in receiving these permissions.
+They will be granted according to the project's need and the requestor's familiarity with cocotb.
+Once you have those permissions, see the guidelines on [managing issues and pull requests](#Managing-of-Issues-and-Pull-Requests).
+
+This task can still be done without special repo permissions.
+The best way to accomplish this is to comment on the issue or PR.
+This is especially helpful for Github issues about bugs.
+If you can duplicate the bug or confirm the bug report is invalid, that helps maintainers *a lot*.
+
+
+### Tests and Coverage
+
+Cocotb has a suite of unit tests (located in the `tests` directory) and examples (located in the `examples` directory) which are functional acceptance tests.
+If a pull request cannot pass *all* of these tests, it will likely be rejected.
+To ensure cocotb only includes the highest quality code, these test should be exhaustive.
+We use code coverage as a quantifiable metric of the "exhaustiveness" of these tests, and wish to improve this metric.
+
+Working this task requires a familiarity with:
+
+* Cocotb's core functionality
+* How to write Verilog and VHDL
+* How to write cocotb tests in Python
+* (Optionally) [codecov](https://docs.codecov.io/docs); coverage aggregator and Github bot
+* (Optionally) the [coverage](https://coverage.readthedocs.io/en/latest/) module, for Python code coverage
+* (Optionally) [gcov](https://gcc.gnu.org/onlinedocs/gcc/Gcov.html), for C++ code coverage
+* (Optionally) [Github Actions](https://docs.github.com/en/free-pro-team@latest/actions), for automatic acceptance testing
+
+Cocotb's regression tests can be improved by:
+
+* Testing more of cocotb's core functionality
+* Testing corner cases left out of the current set of tests (identified by looking at code coverage)
+* Increasing the matrix of simulators, operating system, and Python installations tested in CI
+
+Testing improvements don't require maintainer pre-approval, but require a pull request.
+Please see the [guidelines on submitting pull requests](#patch-requirements).
+
+
+### Features
+
+Cocotb is still in development and new features are still welcome and appreciated;
+as long as they stay [in scope](#Architecture-and-Scope-of-Cocotb).
+Cocotb is comprised of several major codebases, each requiring different sets of skills and development process.
+Instead of including that breakdown here, it is done in the [internal documentation](https://github.com/cocotb/cocotb/wiki/cocotb-Internals).
+
+Small improvements to existing features generally do not require maintainer pre-approval.
+Large changes, approximately >150 LOC changed, and new features generally require maintainer pre-approval.
+
```

Comment:
This section should say something about that a feature can potentially also be developed as an extension.